src/libsyntax/parse/parser.rs

```diff
@@ -5596,26 +5610,77 @@ impl<'a> Parser<'a> {
                 last_comma_span = Some(self.prev_span);
             }
         }
-        if !bad_lifetime_pos.is_empty() {
-            let mut err = self.struct_span_err(
+
+        let mut err = if !bad_lifetime_pos.is_empty() && !bad_type_pos.is_empty() {
```

Comment:
Could we move this logic to its own function? It's starting to get long and interfere with the readability. It can be replaced with something along the lines of `self.maybe_report_incorrect_generic_argument_order(bad_lifetime_pos, bad_type_pos, lifetime_suggestions, type_suggestions);`

Reply:
Fixed.